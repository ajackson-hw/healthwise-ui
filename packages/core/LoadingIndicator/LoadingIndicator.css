.wrapper {
  display: block;
  width: 100%;
  box-sizing: border-box;
  min-height: 50px;
  position: relative;
}

.LoadingIndicator {
  display: inline-block;
  box-sizing: border-box;
  width: 30px;
  height: 30px;
  position: absolute;
  border: 4px solid var(--color-primary-dark);
  top: 10px;
  left: 50%;
  transform: translateX(-15px);
  animation: hw_loader 4s infinite ease;
}

.inner {
  vertical-align: top;
  display: inline-block;
  width: 100%;
  background-color: var(--color-primary-dark);
  animation: hw_loader_inner 4s infinite ease-in;
}

.inverted .LoadingIndicator {
  border-color: var(--color-primary-light);
}

.inverted .inner {
  background-color: var(--color-primary-light);
}

@keyframes hw_loader {
  0% { transform: rotate(0deg); }
  25% { transform: rotate(180deg); }
  50% { transform: rotate(180deg); }
  75% { transform: rotate(360deg); }
  100% { transform: rotate(360deg); }
}

@keyframes hw_loader_inner {
  0% { height: 0%; }
  25% { height: 0%; }
  50% { height: 100%; }
  75% { height: 100%; }
  100% { height: 0%; }
}
