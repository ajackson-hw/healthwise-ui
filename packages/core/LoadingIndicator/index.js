import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import ScreenReaderOnly from '../ScreenReaderOnly'
import styles from './LoadingIndicator.css'

class LoadingIndicator extends React.Component {
  render() {
    const { hiddenText, inverted, ...otherAttributes } = this.props
    const wrapperClass = classNames({
      'hw-loading-indicator-wrapper': true,
      [styles.wrapper]: true,
      [styles.inverted]: inverted,
    })

    return (
      <div className={wrapperClass} {...otherAttributes}>
        <ScreenReaderOnly>{hiddenText}</ScreenReaderOnly>
        <span className={`hw-loading-indicator ${styles.LoadingIndicator}`}>
          <span className={`hw-loading-indicator-inner ${styles.inner}`} />
        </span>
      </div>
    )
  }
}

LoadingIndicator.propTypes = {
  hiddenText: PropTypes.string,
  inverted: PropTypes.bool,
}

LoadingIndicator.defaultProps = {
  hiddenText: 'Loading page',
  inverted: false,
}

export default LoadingIndicator
