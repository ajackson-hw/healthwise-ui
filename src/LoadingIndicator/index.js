import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes, withTheme } from 'styled-components'

import { defaultTheme } from '../Theme'
import ScreenReaderOnly from '../ScreenReaderOnly'

const spinnerOuterAnimation = keyframes`
  0% { transform: rotate(0deg); }
  25% { transform: rotate(180deg); }
  50% { transform: rotate(180deg); }
  75% { transform: rotate(360deg); }
  100% { transform: rotate(360deg); }
`

const spinnerInnerAnimation = keyframes`
  0% { height: 0%; }
  25% { height: 0%; }
  50% { height: 100%; }
  75% { height: 100%; }
  100% { height: 0%; }
`

const Root = styled.div`
  min-height: 50px;
  position: relative;
`

const SpinnerOuter = styled.span`
  display: inline-block;
  box-sizing: border-box;
  width: 30px;
  height: 30px;
  position: absolute;
  border: ${props =>
    props.inverted
      ? `4px solid ${props.theme.colorPrimaryLight}`
      : `4px solid ${props.theme.colorPrimaryDark}`};
  top: 10px;
  left: 50%;
  transform: translateX(-15px);
  animation: ${spinnerOuterAnimation} 4s infinite ease;
`

const SpinnerInner = styled.span`
  vertical-align: top;
  display: inline-block;
  width: 100%;
  background-color: ${props =>
    props.inverted ? props.theme.colorPrimaryLight : props.theme.colorPrimaryDark};
  animation: ${spinnerInnerAnimation} 4s infinite ease-in;
`

class LoadingIndicator extends React.Component {
  render() {
    const { hiddenText, inverted, theme, ...otherProps } = this.props

    return (
      <Root className="hw-loading-indicator-wrapper" {...otherProps}>
        <ScreenReaderOnly>{hiddenText}</ScreenReaderOnly>
        <SpinnerOuter className="hw-loading-indicator" inverted={inverted} theme={theme}>
          <SpinnerInner className="hw-loading-indicator-inner" inverted={inverted} theme={theme} />
        </SpinnerOuter>
      </Root>
    )
  }
}

LoadingIndicator.propTypes = {
  hiddenText: PropTypes.string,
  inverted: PropTypes.bool,
  theme: PropTypes.shape({
    colorPrimaryLight: PropTypes.string,
    colorPrimaryDark: PropTypes.string,
  }),
}

LoadingIndicator.defaultProps = {
  hiddenText: 'Loading page',
  inverted: false,
  theme: defaultTheme,
}

export default withTheme(LoadingIndicator)
