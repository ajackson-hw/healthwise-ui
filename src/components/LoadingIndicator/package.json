{
  "name": "@healthwise/loading-indicator",
  "version": "2.0.0",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - Loading indicator component",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@healthwise/screen-reader-only": "^1.1.3",
    "classnames": "^2.2.5",
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "@healthwise/theme": "^4.0.0",
    "react": ">=15.4.2"
  }
}
