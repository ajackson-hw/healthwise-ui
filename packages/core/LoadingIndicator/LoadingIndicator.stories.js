import React from 'react'
import { storiesOf } from '@storybook/react'

import LoadingIndicator from './index'

const styles = {
  invertedContainer: {
    background: '#424242',
  },
}

storiesOf('UI Feedback/Loading Indicator', module)
  .addWithInfo('with defaults', `Demonstrates default rendering of a LoadingIndicator`, () => (
    <LoadingIndicator />
  ))
  .addWithInfo(
    'with hidden text',
    `Demonstrates a LoadingIndicator with custom hidden text`,
    () => (
      <div>
        <LoadingIndicator hiddenText="This is the custom hidden text" />
        Use a screen reader to hear custom hidden text
      </div>
    )
  )
  .addWithInfo('inverted', `Demonstrates an inverted color LoadingIndicator`, () => (
    <div style={styles.invertedContainer}>
      <LoadingIndicator inverted />
    </div>
  ))
