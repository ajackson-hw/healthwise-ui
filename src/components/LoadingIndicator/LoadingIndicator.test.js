/* global describe, it, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'
import LoadingIndicator from './index'

describe('<LoadingIndicator />', () => {
  it('renders correctly with defaults', () => {
    const component = renderer.create(<LoadingIndicator />)
    expect(component).toMatchSnapshot()
  })

  it('renders correctly with hidden text', () => {
    const component = renderer.create(
      <LoadingIndicator hiddenText="this is the text that will be read by " />
    )
    expect(component).toMatchSnapshot()
  })

  it('renders correctly inverted', () => {
    const component = renderer.create(<LoadingIndicator inverted />)
    expect(component).toMatchSnapshot()
  })
})
