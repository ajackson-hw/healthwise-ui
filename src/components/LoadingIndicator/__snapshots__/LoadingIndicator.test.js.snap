// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> renders correctly inverted 1`] = `
<div
  className="hw-loading-indicator-wrapper wrapper inverted"
>
  <div
    className="hw-screen-reader-only ScreenReaderOnly"
  >
    Loading page
  </div>
  <span
    className="hw-loading-indicator LoadingIndicator"
  >
    <span
      className="hw-loading-indicator-inner inner"
    />
  </span>
</div>
`;

exports[`<LoadingIndicator /> renders correctly with defaults 1`] = `
<div
  className="hw-loading-indicator-wrapper wrapper"
>
  <div
    className="hw-screen-reader-only ScreenReaderOnly"
  >
    Loading page
  </div>
  <span
    className="hw-loading-indicator LoadingIndicator"
  >
    <span
      className="hw-loading-indicator-inner inner"
    />
  </span>
</div>
`;

exports[`<LoadingIndicator /> renders correctly with hidden text 1`] = `
<div
  className="hw-loading-indicator-wrapper wrapper"
>
  <div
    className="hw-screen-reader-only ScreenReaderOnly"
  >
    this is the text that will be read by 
  </div>
  <span
    className="hw-loading-indicator LoadingIndicator"
  >
    <span
      className="hw-loading-indicator-inner inner"
    />
  </span>
</div>
`;
